spec.md

```diff
@@ -906,17 +948,18 @@ message ControllerProbeRequest {
   // The API version assumed by the CO. This is a REQUIRED field.
   Version version = 1;
 }
+```
 
-message ControllerProbeResponse {
-  message Result {}
+##### ControllerProbe Errors
 
-  // One of the following fields MUST be specified.
-  oneof reply {
-    Result result = 1;
-    Error error = 2;
-  }
-}
-```
+If the plugin is unable to complete the ControllerProbe call successfully, it MUST return a non-ok gRPC code in the gRPC status.
+If the conditions defined below are encountered, the plugin MUST return the specified gRPC error code.
+The CO MUST implement the specified error recovery behavior when it encounters the gRPC error code.
+
+Condition | gRPC Code | Description | Recovery Behavior
+| --- | --- | --- | --- |
+| Bad plugin config | 3 INVALID_ARGUMENT | Indicates that the plugin is misconfigured. | Caller MUST assume the plugin is not healthy. |
```

Comment:
`INVALID_ARGUMENT` here is weird because we don't really specify any argument for the call :)
I prefer `FailedPrecondition` here.

Reply:
I concur 100%. This is the entire purpose of `FailedPrecondition` as it relates to *the state of the system*.

Reply:
Agreed. Brain farted on that one.